validation/userRegisterValidator.ts

```diff
@@ -0,0 +1,12 @@
+import * as yup from "yup"; 
+
+export const registerSchema = yup.object({
+    userName: yup.string()
+        .trim()
+        .required(),
+    email: yup.string().email().trim().required(),
+    password: yup.string().trim().min(4).max(10)
+        // Minimum eight and maximum 10 characters, at least one uppercase letter, one lowercase letter, one number and one special characte
+        // .matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{4,10}$/)
```

Comment:
As the .matches statement is commented out it is inactive. Is there a reason for this? When this is uncommented it will require users to create strong passwords. Also, we will need more tests to check that this works as intended.

Reply:
That was an example. We don't have any rules about password. I thought after discuss about the password rule, I would change it but if you are happy with the rule, I will uncomment it.

Reply:
I posted it on Discord and it seems most would be happy with the pattern below to begin with. By adding this pattern you no longer need to add .min(4).max(10) as the pattern will also check length.

// Minimum six and maximum 15 characters, at least one uppercase letter, one lowercase letter, one number and one special character
// .matches(/^(?=.[a-z])(?=.[A-Z])(?=.\d)(?=.[-@$!%?&])[A-Za-z\d-@$!%?&]{6,15}$/)